### ISSUE TITLE
Scala 3: incorrect highlighting in `given Type = ...`

### ISSUE THREAD

**Issue by kubukoz** [scalameta/metals#2534]
**Describe the bug**

The highlighting of the code background seems to be wrong in a situation where a `given` is defined without a name:

```scala
trait Eq[A]
given Eq[Int] = ???
```

This appears in the vscode extension if you put the cursor somewhere between these positions (marked by angle brackets):

```
given <Eq[Int>] = ???
```

**To Reproduce**

Write the above example to a file, put the cursor somewhere within `Eq[Int`.

Note: this doesn't happen if the given has a name.

**Expected behavior**

`Eq` and its references are highlighted.

**Screenshots**

![image](https://user-images.githubusercontent.com/894884/108610702-1822dc00-73d8-11eb-8e9d-63746f7cb21a.png)

**Installation:**

- Operating system: macOS
- Editor: Visual Studio Code
- Metals version: v0.9.10+151-aaa761e5-SNAPSHOT

**Additional context**

- Scala version: 3.0.0-M3

**Search terms**

highlighting, givens, implicits, references, symbols

**Comment by dos65** [comment]
Thanks for reporting!
I'm able to reproduce it.

**Comment by tgodzik** [comment]
That is probably another one that we need to fix upstream:
```
src/main/scala/Test.scala
-------------------------
Occurrences:
[0:8..0:9) <= a/
[2:6..2:11) <= a/Other#
[2:11..2:11) <= a/Other#`<init>`().
[2:12..2:13) <= a/Other#[T]
[4:0..4:0) <= a/Test$package.
[4:6..4:11) => a/Other#
[4:6..4:21) <= a/Test$package.given_Other_Int.
[4:12..4:15) => scala/Int#
[4:19..4:22) => scala/Predef.`???`().
```

**Comment by tgodzik** [comment]
This is fixed in `3.0.0-RC2`
